### ISSUE TITLE
Unsupported charset "134": mb_convert_encoding(): Illegal character encoding specified

### ISSUE THREAD

**Issue by fbg13** [ddeboer/imap#270]
Getting this error when calling either `$email->getBodyText();` or `$email->getBodyHtml();` on a specific email.

```
Exception: [E_WARNING] Unsupported charset "134": mb_convert_encoding(): Illegal character encoding specified
imap_alerts (0):
imap_errors (0): (in /home/fbg/www/sol/vendor/ddeboer/imap/src/Message/Transcoder.php line 304)

#0 /home/fbg/www/sol/vendor/ddeboer/imap/src/Message/AbstractPart.php(290): Ddeboer\Imap\Message\Transcoder::decode('<!DOCTYPE HTML ...', '134')
#1 /home/fbg/www/sol/vendor/ddeboer/imap/src/Message/AbstractMessage.php(181): Ddeboer\Imap\Message\AbstractPart->getDecodedContent()
```

Email source (removed personal information) https://pastebin.com/50ARcrd9

**Comment by artemevsin** [comment]
after some googling, found this:
https://msdn.microsoft.com/en-us/library/cc194829.aspx
https://en.wikipedia.org/wiki/GB_2312

mb_convert_encoding() maybe support this encoding http://php.net/manual/en/mbstring.supported-encodings.php#113217 , but two minuses do not sound good. 

Anyway, I do not see any chinese characters there, also sender's IP address is in USA, so probably it is just bad encoding definition

**Comment by Slamdunk** [comment]
`GB2312` is supported both by `iconv` and `mbstring` as an alias of `EUC-CN`, so it is not an issue itself.

For sure the issue is that the software that generated the email didn't convert `134` to `GB2312`, but we know that email ecosystem is a mess, so we need to support it anyway 😦 

The page that @artemevsin cited, https://msdn.microsoft.com/en-us/library/cc194829.aspx (and others similar in microsoft site) is the only source I found of the possible candidate for the right charset of 134: I'll create a PR soon to support them as well.

**Comment by Slamdunk** [comment]
Released in [1.1.2](https://github.com/ddeboer/imap/releases/tag/1.1.2)
